docs/packages/rapids_cpm_versions.rst

```diff
@@ -0,0 +1,63 @@
+:orphan:
+
+
+rapids-cmake package version format
+###################################
+
+
+rapids-cmake uses a JSON file to encode the version of a project and how to download the project.
+
+The JSON format is a root object that contains the ``packages`` object.
+
+The ``packages`` object contains a key/value map of all supported
+packages where the key is the case senstive name of the project and
+the value is a ``project`` object, as seen in this example:
+
+.. literalinclude:: /packages/example.json
+  :language: json
+
+
+Project Object Fields
+*********************
+
+Each ``project`` object must contain the following fields so that
+rapids-cmake can properly use CPM to find or download the project
+as needed.
+
+``version``
+
+    A required string representing the version of the project to be used
+    by :cmake:command:`rapids_cpm_find` when looking for a local installed
+    copy of the project.
+
+    Supports the following placeholders:
+        - ``${rapids-cmake-version}`` will be evulated to 'major.minor' of the current rapids-cmake cal-ver value.
+
+``git_url``
+
+    A required string representing the git url to be used when cloning the
+    project locally by the :cmake:command:`rapids_cpm_find` when a locally
+    installed copy of the project can't be found.
+
+``git_tag``
+
+    A required string representing the git tag to be used when cloning the
+    project locally by the :cmake:command:`rapids_cpm_find` when a locally
+    installed copy of the project can't be found.
+
+    Supports the following placeholders:
+        - ``${rapids-cmake-version}`` will be evulated to 'major.minor' of the current rapids-cmake cal-ver value.
+        - ``${version}`` will be evulated to the contents of the ``version`` field.
+
+``git_shallow``
+
+    An optional boolean value that represents if we should do a shallow git clone
+    or not. If no such field exists the default is `git_shallow : true`
+
```

Comment:
I think we need the ability to patch things as well, i.e. https://github.com/rapidsai/cudf/blob/branch-21.10/cpp/cmake/thirdparty/CUDF_GetThrust.cmake#L33

Reply:
I think that is worth doing as a followup Issue/Feature discussion and PR. 

As you are aware since cudf patches thrust it doesn't install it in the default location, which is what this change does. I am hesitant to install a modified Thrust in the 'default' location as that could cause confusion to users ( potential compile time or performance differences ). But once we resolve where patched versions of projects should be installed, this will be required.